spring-data-mongodb/src/main/java/org/springframework/data/mongodb/observability/MongoObservation.java

```diff
@@ -172,6 +175,15 @@ public String asString() {
 			public String asString() {
 				return "db.operation";
 			}
+		};
+
+		/**
+		 * Creates a key value for the given key name.
+		 * @param value value for key, if value is null or empty {@link KeyValue.NONE_VALUE} will be used
+		 * @return key value
+		 */
+		public KeyValue withOptionalValue(@Nullable String value) {
+			return withValue(ObjectUtils.isEmpty(value) ? KeyValue.NONE_VALUE : value);
```

Comment:
I wonder if "UNKNOWN" is a better default value for most of these rather than none.

Reply:
@shakuzen 
>I wonder if "UNKNOWN" is a better default value for most of these rather than none.

I think so, I can also make the fallback mandatory so it needs to be decided for every key-value.

@mp911de
>I would like to have a bit more encapsulated API for a more declarative approach [...]

We can consider improving on this in Micrometer 1.16 (let me talk to @shakuzen about it) but I think this should be considered as a bug and should be fixed in earlier versions where Micrometer 1.16 will not be available I think (unless you want to copy some code from there).

Reply:
https://github.com/micrometer-metrics/micrometer/issues/6564